src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -69,61 +67,25 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
       state.acquisitionData
     )
 
-  private def buildSubscribeRequest(state: CreateZuoraSubscriptionState) = {
+  private def buildSubscribeRequest(state: CreateZuoraSubscriptionState, promotionService: PromotionService) = {
     //Documentation for this request is here: https://www.zuora.com/developer/api-reference/#operation/Action_POSTsubscribe
     SubscribeItem(
       buildAccount(state),
       buildContactDetails(state),
       state.paymentMethod,
-      buildSubscriptionData(state),
+      buildSubscriptionData(state, promotionService),
       SubscribeOptions()
     )
   }
 
-  private def buildSubscriptionData(state: CreateZuoraSubscriptionState) = {
+  private def buildSubscriptionData(state: CreateZuoraSubscriptionState, promotionService: PromotionService) = {
     val config = zuoraConfigProvider.get(state.user.isTestUser)
     state.product match {
-      case c: Contribution =>
-        val contributionConfig = config.contributionConfig(c.billingPeriod)
-        buildProductSubscription(
-          contributionConfig.productRatePlanId,
-          List(
-            RatePlanChargeData(
-              RatePlanCharge(contributionConfig.productRatePlanChargeId, Some(c.amount)) //Pass the amount the user selected into Zuora
-            )
-          )
-        )
-      case d: DigitalPack =>
-        val contractEffectiveDate = LocalDate.now(DateTimeZone.UTC)
-        val contractAcceptanceDate = contractEffectiveDate
-          .plusDays(config.digitalPack.defaultFreeTrialPeriod)
-          .plusDays(config.digitalPack.paymentGracePeriod)
-
-        buildProductSubscription(
-          config.digitalPackRatePlan(d.billingPeriod),
-          contractAcceptanceDate = contractAcceptanceDate,
-          contractEffectiveDate = contractEffectiveDate
-        )
+      case c: Contribution => c.build(config)
```

Comment:
Extracted to a separate class `ProductSubscriptionBuilders` as it was getting quite complex